Jetpack Section: Add Warning and Restore Status screens

momo-ozawa opened the pull request:
Part of #15191 

- Adds Warning skeleton screen
- Adds Restore Status skeleton screen

⚠️ **The following will be addressed in subsequent PRs:**
- Passing data
- API hookup
- Styling

### To test:

1. My Site > Activity Log > rewindable activity cell
2. Tap on ellipsis icon
3. Tap on `Restore`
4. Tap on `Restore to this point`
     - ✅ Should display new Warning screen
 5. Tap on `Confirm`
     - ✅ Should display new Restore Status screen

Warning | Restore Status
--- | ---
![Screen Shot 2021-01-07 at 20 48 57](https://user-images.githubusercontent.com/6711616/103889892-7d718500-512a-11eb-81c8-17eb8cc572da.png) | ![Screen Shot 2021-01-07 at 20 49 00](https://user-images.githubusercontent.com/6711616/103889897-7fd3df00-512a-11eb-9ff9-a15e80aee1e1.png)


### PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Commit:
Add Jetpack Restore Warning screen

Commit:
Add actionButtonHandler for action button tapped

Commit:
Extract strings to enum

Commit:
Add Jetpack Restore Progress screen

Commit:
Update styling for header view

Commit:
Add RestoreWarningView

Commit:
Add RestoreStatusView

Commit:
Reorganize Jetpack Restore files

Commit:
Define button handler closures

leandroalonso commented on WordPress/Classes/ViewRelated/Jetpack/Jetpack Restore/Restore Status/RestoreStatusView.swift:
```diff
@@ -0,0 +1,80 @@
+    // MARK: - Styling
+
+    private func applyStyles() {
+        backgroundColor = .white
```
For light/dark mode:

```suggestion
        backgroundColor = .basicBackground
```

leandroalonso commented on WordPress/Classes/ViewRelated/Jetpack/Jetpack Restore/Restore Warning/RestoreWarningView.swift:
```diff
@@ -0,0 +1,67 @@
+    // MARK: - Styling
+
+    private func applyStyles() {
+        backgroundColor = .white
```
For light/dark mode:

```suggestion
        backgroundColor = .basicBackground
```

leandroalonso reviewed (commented):
@momo-ozawa works as described, I spotted a few issues:

* There are some hardcoded `white` colors that break the dark mode
* When I increased the font size in Simulator, I wasn't able to tap the "Restore to this point" button. Not sure if it's just a local problem tho

Also, should we change the event detail to trigger this flow? The "Rewind" button:

<img src="https://user-images.githubusercontent.com/7040243/103898290-7b300c00-50d3-11eb-98dd-655fbf000e04.png" width="300">

Commit:
Update routing for rewind button on Activity Detail

Commit:
Update styling

Commit:
Fix trailing whitespace

momo-ozawa commented:
@leandroalonso 

> There are some hardcoded white colors that break the dark mode
>
> When I increased the font size in Simulator, I wasn't able to tap the "Restore to this point" button. Not sure if it's just a local problem tho

Fixed! 34bf558

> Also, should we change the event detail to trigger this flow? The "Rewind" button:

Fixed! fce71bb 

Restore Options | Warning | Restore Status
--- | --- | ---
![Screen Shot 2021-01-08 at 13 48 19](https://user-images.githubusercontent.com/6711616/103976899-58792280-51bb-11eb-950f-78ad7e9a1599.png) | ![Screen Shot 2021-01-08 at 13 48 22](https://user-images.githubusercontent.com/6711616/103976900-59aa4f80-51bb-11eb-9b85-6b8014b7e525.png) | ![Screen Shot 2021-01-08 at 13 48 25](https://user-images.githubusercontent.com/6711616/103976902-5a42e600-51bb-11eb-8bd0-6005d15ae2c7.png)
